tensorwaves/physics/helicityformalism/amplitude.py

```diff
@@ -501,56 +504,104 @@ def __call__(self, dataset: dict) -> tf.Tensor:
         ) * self._dynamics_function(dataset)
 
 
+class _HelicityParticle:
+    def __init__(self, name: str, helicity: int,) -> None:
+        self.name: str = name
+        self.helicity: int = helicity
+
+    @staticmethod
+    def from_dict(definition: Dict[str, Any]) -> "_HelicityParticle":
+        name = str(definition["Name"])
+        helicity = int(definition["Helicity"])
+        return _HelicityParticle(name, helicity)
+
+
+class _DecayProduct(_HelicityParticle):
+    def __init__(
+        self, name: str, helicity: int, final_state_ids: List[int],
+    ) -> None:
+        super().__init__(name, helicity)
+        self.final_state_ids: List[int] = final_state_ids
+
+    @staticmethod
+    def from_dict(definition: Dict[str, Any]) -> "_DecayProduct":
+        helicity_particle = _HelicityParticle.from_dict(definition)
+        final_state_ids = _safe_wrap_list(definition["FinalState"])
+        return _DecayProduct(
+            helicity_particle.name, helicity_particle.helicity, final_state_ids
+        )
+
+
+class _RecoilSystem:
+    def __init__(
+        self,
+        recoil_ids: Optional[List[int]] = None,
+        parent_recoil_ids: Optional[List[int]] = None,
+    ) -> None:
+        self.recoil_ids: List[int] = []
+        self.parent_recoil_ids: List[int] = []
+        if recoil_ids:
+            self.recoil_ids = recoil_ids
+        if parent_recoil_ids:
+            self.parent_recoil_ids = parent_recoil_ids
+
+    @staticmethod
+    def from_dict(definition: Dict[str, Any]) -> "_RecoilSystem":
+        recoil_system = _RecoilSystem()
+        if "RecoilFinalState" in definition["RecoilSystem"]:
+            recoil_system.recoil_ids = _safe_wrap_list(
+                definition["RecoilSystem"]["RecoilFinalState"]
+            )
+        if "ParentRecoilFinalState" in definition["RecoilSystem"]:
+            recoil_system.parent_recoil_ids = _safe_wrap_list(
+                definition["RecoilSystem"]["ParentRecoilFinalState"]
+            )
+        return recoil_system
+
+
```

Comment:
So this part goes a bit into #77. It doesn't fix that issue, but it's an attempt at formulating a data structure that can then be filled/created from a recipe file (`dict`).

Anyway, in this PR the motivation for creating these classes was just to reduce the number of local variables under `_create_helicity_decay` that `pylint` was complaining about.

Reply:
Oh sorry, so that comment too late. All right.